### DISCUSSION TITLE
GitHub runners physical location

### DISCUSSION THREAD

**Discussion by michalbali256** [community/community#24969] [Actions]
<p>Hi,</p>
<p>My company has an obligation to specify country, where our software was built. In which countries are the action runners hosted? Is there a way to determine the location of current runner? Is there a way to restrict the countries in which my runners execute?</p>
<p>Thanks</p>

**Comment by otter-computer** [comment, marked as answer]
<p>Hi <a class="mention" href="/michalbali256">@michalbali256</a>, welcome to the GitHub Support Community! GitHub hosts runners in the following Azure regions:</p>
<ul>
<li>East US (eastus)</li>
<li>East US 2 (eastus2)</li>
<li>West US 2 (westus2)</li>
<li>Central US (centralus)</li>
<li>South Central US (southcentralus)</li>
</ul>
<p>You can compare these regions with the Azure data centre locations here:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://azurecomcdn.azureedge.net/cvt-c9656ea1b8baa351e4f9db7bc9652db00de5177052aef1b1ef5cf0e5b10af08c/images/icon/favicon.ico" class="site-icon" width="128" height="128">
      <a href="https://azure.microsoft.com/en-us/global-infrastructure/geographies/" target="_blank" rel="noopener">azure.microsoft.com</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/2439407/181087950-72d32919-8607-42d2-a48f-6eeed3b8da52.png" class="thumbnail onebox-avatar" width="250" height="250">

<h3><a href="https://azure.microsoft.com/en-us/global-infrastructure/geographies/" target="_blank" rel="noopener">Choose the Right Azure Region for You | Microsoft Azure</a></h3>

<p>Azure has more geographies than any other cloud provider—with robust data sovereignty, residency, compliance, and resiliency capabilities for any application.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>You may also be able to find what you need by checking the IPs of the GitHub runners here:</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/actions/reference/specifications-for-github-hosted-runners#ip-addresses" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/actions/reference/specifications-for-github-hosted-runners#ip-addresses" target="_blank" rel="noopener">Specifications for GitHub-hosted runners - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/actions/reference/specifications-for-github-hosted-runners#ip-addresses" target="_blank" rel="noopener">//docs.github.com/en/actions/reference/specifications-for-github-hosted-runners</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>You aren’t able to choose which runner locations are used though so if all these locations don’t suit your needs then you may need to find an alternate solution.</p>
<p>If you need to be more specific about where your code is processed, then it may be worth looking into setting up self-hosted runners in your chosen location:</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners" target="_blank" rel="noopener">About self-hosted runners - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners" target="_blank" rel="noopener">//docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Reply by ryangoree** [reply]
Is this list maintained anywhere in the docs? I imagine it's subject to change?

**Reply by Nfsaavedra** [reply]
Are the regions used still the same?

**Reply by Nfsaavedra** [reply]
https://github.com/Nfsaavedra/github-actions-location-test
